scheduler/src/cook/mesos/util.clj

```diff
@@ -91,22 +92,35 @@
 (defn entity->map
   "Takes a datomic entity and converts it along with any nested entities
   into clojure maps"
-  ([entity & {:keys [db skip-keys]
-              :or {skip-keys #{}}}]
-   (let [entity' (if db (d/entity db (:db/id entity))
-                     entity)]
-     (deep-transduce-kv (filter (fn [[k _]] (not (contains? skip-keys k)))) entity'))))
-
-(defn job-ent->map
-  "Convert a job entity to a map. This also loads the associated group and converts it to a map."
-  ([job db]
-   (job-ent->map (d/entity db (:db/id job))))
-  ([job]
-   (-> (entity->map job)
-       ; Load the group related to the job (if any) but without loading the job entities
-       ; inside the group entity. If someone needs to access them, it's better to query
-       ; at that point instead of relying on the cached data.
-       (assoc :group/_job (entity->map (:group/_job job) :skip-keys #{:group/job})))))
+  ([entity db]
+   (entity->map (d/entity db (:db/id entity))))
+  ([entity]
+   (deep-transduce-kv (map identity) entity)))
+
+; During testing, we found that calling (keys) on a group entity with a large number of jobs was
+; quite slow. As an optimization, load the group keys ahead of time from the schema.
```

Comment:
question: Do we know why that call is slow? What is it iterating over to load those keys?

Reply:
I don't know, without digging into datomic I'm not really sure. I wanted to get this wrapped up but I can spend some more time digging into it.

Reply:
We can do it outside this PR and update the comment later.

Reply:
Can we quantify "quite slow"?

Reply:
Added a comment for posterity